src/squlearn/qnn/base_qnn.py

```diff
@@ -108,11 +121,85 @@ def fit(self, X: np.ndarray, y: np.ndarray, weights: np.ndarray = None) -> None:
             y: Labels
             weights: Weights for each datapoint
         """
-        self.param = self.param_ini.copy()
-        self.param_op = self.param_op_ini.copy()
+        self._param = self.param_ini.copy()
+        self._param_op = self.param_op_ini.copy()
         self._is_fitted = False
         self._fit(X, y, weights)
 
+    def get_params(self, deep: bool = True) -> dict:
+        """
+        Returns a dictionary of parameters for the current object.
+
+        Parameters:
+            deep: If True, includes the parameters from the base class.
+
+        Returns:
+            dict: A dictionary of parameters for the current object.
+        """
+        # Create a dictionary of all public parameters
+        params = super().get_params(deep=False)
+
+        if deep:
+            params.update(self._qnn.get_params(deep=True))
+        return params
+
+    def set_params(self: BaseQNN, **params) -> BaseQNN:
+        """
+        Sets the hyper-parameters of the BaseQNN.
+
+        Args:
+            params: Hyper-parameters of the BaseQNN.
+
+        Returns:
+            updated BaseQNN
+        """
+        # Create dictionary of valid parameters
+        valid_params = self.get_params().keys()
+        for key in params.keys():
+            # Check if parameter is valid
+            if key not in valid_params:
+                raise ValueError(
+                    f"Invalid parameter {key!r}. "
+                    f"Valid parameters are {sorted(valid_params)!r}."
+                )
+
+        # Set parameters
+        self_params = params.keys() & self.get_params(deep=False).keys()
+        for key in self_params:
+            setattr(self, key, params[key])
+
+        # Set parameters of the QNN
+        qnn_params = params.keys() & self._qnn.get_params(deep=True).keys()
+        if qnn_params:
+            self._qnn.set_params(
+                **{key: value for key, value in params.items() if key in qnn_params}
+            )
+
+            # If the number of parameters has changed, reinitialize the parameters
+            if self.feature_map.num_parameters != len(self.param_ini):
+                self.param_ini = self.feature_map.generate_initial_parameters(
+                    seed=self.parameter_seed
+                )
+            if isinstance(self.operator, list):
+                num_op_parameters = sum(operator.num_parameters for operator in self.operator)
+                if num_op_parameters != len(self.param_op_ini):
+                    self.param_op_ini = np.concatenate(
+                        [
+                            operator.generate_initial_parameters(seed=self.parameter_seed)
+                            for operator in self.operator
+                        ]
+                    )
+            elif self.operator.num_parameters != len(self.param_op_ini):
```

Comment:
same here, you can get the number of parameters and operator_parameters also from the qnn class.